sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -4020,6 +4020,45 @@ object GpuOverrides extends Logging {
           GpuDynamicPruningExpression(child)
         }
       }),
+    expr[HyperLogLogPlusPlus](
+      "Aggregation approximate count distinct",
+      ExprChecks.reductionAndGroupByAgg(TypeSig.LONG, TypeSig.LONG,
+        // HyperLogLogPlusPlus depends on Xxhash64
+        // HyperLogLogPlusPlus supports all the types that Xxhash 64 supports
+        Seq(ParamCheck("input",XxHash64Shims.supportedTypes, TypeSig.all))),
+      (a, conf, p, r) => new UnaryExprMeta[HyperLogLogPlusPlus](a, conf, p, r) {
+
+        // It's the same as Xxhash64
+        override def tagExprForGpu(): Unit = {
+          val maxDepth = a.children.map(
+            c => XxHash64Utils.computeMaxStackSize(c.dataType)).max
+          if (maxDepth > Hash.MAX_STACK_DEPTH) {
+            willNotWorkOnGpu(s"The data type requires a stack depth of $maxDepth, " +
+                s"which exceeds the GPU limit of ${Hash.MAX_STACK_DEPTH}. " +
+                "The algorithm to calculate stack depth: " +
+                "1: Primitive type counts 1 depth; " +
+                "2: Array of Structure counts:  1  + depthOf(Structure); " +
+                "3: Array of Other counts: depthOf(Other); " +
+                "4: Structure counts: 1 + max of depthOf(child); " +
+                "5: Map counts: 2 + max(depthOf(key), depthOf(value)); "
+            )
+          }
+          // Spark guarantee the precision is bigger or equal to 4
+          val precision = GpuHyperLogLogPlusPlus.computePrecision(a.relativeSD)
+          // Spark-Rapids only supports precision <= 15
+          // https://github.com/NVIDIA/spark-rapids/issues/12347
+          if (precision > 14) {
+            // cuCollection only supports precision range [4, 18]
```

Comment:
`[4, 18]` or `[4, 14]`?

Reply:
// Info: cuCollection supports precision range [4, 18]
            // Due to https://github.com/NVIDIA/spark-rapids/issues/12347, the Spark-Rapids supports
            // fewer precisions than cuCollection: range: [4, 14]